### DISCUSSION TITLE
播放加密视频

### DISCUSSION THREAD

**Discussion by nzm001** [hooke007/mpv_PlayKit#118] [mpv 通用]
由于这几年来已经习惯于抓后浪站（码率真的低）的视频链接放到mpv里直接播放，然后配上相应着色器，观感勉强还行。不过目前由于部分正版看不了，发现现在的dao版网站用的m3u8格式，放到mpv无法播放。

请问大佬，mpv如何播放加密过的视频。以m3u8格式为例，如果我本地有key文件，播放的命令该如何写（mpv的issues上有一两个讨论，但是我看完全不懂。。）。

另外，请问各位大佬有没有比较优雅开源的方案提取key（油猴浏览器插件都没找到）。

具体案例视频：[http://www.dmh8.com/player/8526-0-0.html](http://www.dmh8.com/player/8526-0-0.html)，[https://ysgc.me/vodplay/48992-2-1.html](https://ysgc.me/vodplay/48992-2-1.html)。想用mpv播放其中的视频（不下载视频到本地的情况下）。

**Comment by kix99aug** [comment]
這是我使用mpv撥放巴哈姆特動畫瘋所使用的小工具，你可以參考裡面的寫法
https://github.com/kix99aug/ani.gamer_url_handler

**Reply by nzm001** [reply]
谢谢，这个可以直接看，我就不折腾了。巴哈姆特好像不是加密的m3u8。对付加密我目前只能下载m3u8和自己根据浏览器f12去生成key，然后编辑m3u8文件：`#EXT-X-KEY:METHOD=AES-128,URI="D:/我的key"`。

**Reply by kix99aug** [reply]
![image](https://user-images.githubusercontent.com/10905628/168442545-2a8c8916-064f-40b1-94e0-2e33a196635f.png)
巴哈是有加密的 可以直接撥放
